content/lessons/3_data_types/data-types.rs

```diff
@@ -0,0 +1,57 @@
+#[derive(Debug)]
+struct Position(i32, i32); // tuple struct
+
+#[derive(Debug)]
+struct Hero {
+    name: String,
+    level: u32,
+    experience: u32,
+    position: Position,
+}
+
+// we can add methods to structs using the 'impl' keyword
+impl Hero {
+    // static method
+    fn new(name: String) -> Hero {
+        Hero {
+            name,
+            level: 1,
+            experience: 0,
+            position: Position(0, 0),
+        }
+    }
+}
+
+// multiple impl blocks are possible for one struct
+impl Hero {
+    // instance method, first argument (self) is the calling instance
+    fn get_distance(&self, pos: Position) -> Position {
+        // fields of tuples and tuple structs can be accessed through 'tuple.[i]'
+        Position(pos.0 - self.position.0, pos.1 - self.position.1)
+    }
+
+    // mutable borrow of self allows to change instance fields
+    fn level_up(&mut self) {
+        self.experience = 0;
+        self.level += 1;
+    }
+}
+
+fn main() {
+    let mut hero: Hero = Hero::new(String::from("Marty The Brave"));
+    hero.level_up(); // 'self' is always passed implicitly
+
+    // fields other than 'name' will be the same as in 'hero'
+    let brother = Hero {
```

Comment:
this is nice, but maybe it's a good moment to introduce #[derive(Clone)] as well?